src/Docker.PowerShell/Cmdlets/GetContainer.cs

```diff
@@ -1,25 +1,56 @@
 ﻿using System.Management.Automation;
 using Docker.DotNet.Models;
 using System.Threading.Tasks;
+using Docker.PowerShell.Objects;
 
 namespace Docker.PowerShell.Cmdlets
 {
     [Cmdlet(VerbsCommon.Get, "Container",
-            DefaultParameterSetName = CommonParameterSetNames.Default)]
+            DefaultParameterSetName = CommonParameterSetNames.ContainerName)]
     [OutputType(typeof(ContainerListResponse))]
     public class GetContainer : DkrCmdlet
     {
+        [Parameter(ParameterSetName = CommonParameterSetNames.Default,
+            ValueFromPipeline = true,
+                   Position = 0)]
+        [ValidateNotNullOrEmpty]
+        [ArgumentCompleter(typeof(ContainerArgumentCompleter))]
+        public string[] Id { get; set; }
+
+        [Parameter(ParameterSetName = CommonParameterSetNames.ContainerName,
+            ValueFromPipeline = true,
+                   Position = 0)]
+        [ValidateNotNullOrEmpty]
+        [ArgumentCompleter(typeof(ContainerArgumentCompleter))]
+        public string[] Name { get; set; }
 
         #region Overrides
         /// <summary>
         /// Outputs container objects for each container matching the provided parameters.
         /// </summary>
         protected override async Task ProcessRecordAsync()
-        {
-            foreach (var c in await DkrClient.Containers.ListContainersAsync(
-                new ContainersListParameters() { All = true }))
+        {            
+            if (Id != null)
+            {
+                foreach (var id in Id)
+                {
+                    WriteObject(await ContainerOperations.GetContainersById(id, DkrClient));    
```

Comment:
I think this is safe; the underlying function returns an array that happens to be empty, as we discussed offline.